### PR TITLE
Filebeat improvements

### LINKED ISSUE #558: Filebeat improvements
After testing the Helm chart for filebeat in cooperation with the Integrations team (thank you @ChrsMark), we detected the following list of items to be updated in the chart:

- [ ] Remove `in_cluster` from add_kubernetes_metadata (partially removed [here](https://github.com/elastic/helm-charts/pull/403)): 
  - Examples: https://github.com/elastic/helm-charts/blob/9b406f8acc84870022dccd123c6ed9018d3cfc6b/filebeat/examples/security/values.yaml#L9
- [ ] File inputs Type is different
  - Helm chart: https://github.com/elastic/helm-charts/blob/a03b78e40d5f4c4b71a1e4e9ca08ffd9191bb700/filebeat/values.yaml#L7
  -  Filebeat reference: https://github.com/elastic/beats/blob/7.6/deploy/kubernetes/filebeat-kubernetes.yaml#L12-L15
- [ ] Kubernetes metadata uses defaults.
  - Helm chart: https://github.com/elastic/helm-charts/blob/a03b78e40d5f4c4b71a1e4e9ca08ffd9191bb700/filebeat/values.yaml#L11
  - Filebeat reference: https://github.com/elastic/beats/blob/7.6/deploy/kubernetes/filebeat-kubernetes.yaml#L16-L20

cc/ @jmlrt

### PR DESCRIPTION
**Pull request by jmlrt**
- Remove `in_cluster` config from `add_kubernetes_metadata
- Use `container` input instead of `docker` input
- Use recommended values for `add_kubernetes_metadata`

Fixes #558 

- [x] Chart version *not* bumped (the versions are all bumped and released at the same time)
- [ ] ~README.md updated with any new values or changes~
- [ ] ~Updated template tests in `${CHART}/tests/*.py`~
- [x] Updated integration tests in `${CHART}/examples/*/test/goss.yaml`

### CODE REVIEW

File: filebeat/examples/security/values.yaml [hunk 1]
```diff
@@ -1,12 +1,15 @@
 filebeatConfig:
   filebeat.yml: |
     filebeat.inputs:
-    - type: docker
-      containers.ids:
-      - '*'
+    - type: container
+      paths:
+        - /var/log/containers/*.log
```
Comment (ChrsMark) [inline filebeat/examples/security/values.yaml:6]: Just to be sure, is this path actually where hosts' `var/log` is mounted? This is what I expect to see https://github.com/elastic/beats/blob/1a029e547fb1018668e7ab961bcbbe496696ff25/deploy/kubernetes/filebeat-kubernetes.yaml#L110.

Comment (jmlrt) [reply filebeat/examples/security/values.yaml:6]: Thanks for the notice, fixed by https://github.com/elastic/helm-charts/pull/568/commits/72f704ed43f0a04a84cc778ece51acdb83a28b9e

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by jmlrt** [comment]
jenkins test this please

File: filebeat/examples/6.x/values.yaml [hunk 2]
```diff
@@ -3,3 +3,16 @@ imageTag: 6.8.8
...
+    filebeat.inputs:
+    - type: docker
+      containers.ids:
+        - '*'
+      processors:
+      - add_kubernetes_metadata:
+          in_cluster: true
+    output.elasticsearch:
+      host: '${NODE_NAME}'
+      hosts: '${ELASTICSEARCH_HOSTS:elasticsearch-master:9200}'
```
Comment (mgreau) [inline filebeat/examples/6.x/values.yaml:18]: can you add a new line please

**Pull request merged**